packages/rest/src/rest.server.ts

```diff
@@ -572,24 +580,7 @@ export class RestServer extends Context implements Server, HttpServerLike {
     // Setup the HTTP handler so that we can verify the configuration
     // of API spec, controllers and routes at startup time.
     this._setupHandlerIfNeeded();
-
-    const httpPort = await this.get<number>(RestBindings.PORT);
-    const httpHost = await this.get<string | undefined>(RestBindings.HOST);
-    this._httpServer = createServer(this.requestHandler);
```

Comment:
Continuing my previous comment - this is the place where to instantiate `HttpServer` as a replacement for `http.createServer`. We need to preserve the logic obtaining the latest values for port & host.

Reply:
Cross-posting https://github.com/strongloop/loopback-next/pull/1369#discussion_r191401296

An outline of a test case to write:

```ts
const server = new RestServer({port: 80});
server.bind(RestBindings.PORT).to(0);
await server.start();
// get the actual port number and verify it's not 80
```